app/frontend/src/features/userQueries/useCurrentUser.ts

```diff
@@ -0,0 +1,8 @@
+import { useAuthContext } from "../auth/AuthProvider";
+import { useUser } from "./useUsers";
+
+export default function useCurrentUser() {
+  const authState = useAuthContext().authState;
+  ///TODO: is this assert okay?
```

Comment:
note

Reply:
I guess this could error unless it can be guaranteed to only be used in authenticated routes?

Reply:
Yes, not sure how to guarantee beyond throwing though!

Reply:
That sounds like a reasonable approach to me, similar to what you did for `useAuthContext`. At least future devs/us who use it wrong will get feedback quickly that way.

Reply:
So let's make this throw an error if `authState.userId` is null/undefined?

Reply:
I didn't realize `!.` compiles to `.` in TS!

Reply:
Yeah I just read that too. I guess it's an escape hatch to say TS should trust what we're doing and if anything goes wrong at run time, it's our fault :sweat_smile: